cmd/finch/nerdctl.go

```diff
@@ -215,3 +243,99 @@ var cmdFlagSetMap = map[string]map[string]sets.Set[string]{
 		"shortArgFlags": sets.New[string]("-e", "-h", "-m", "-u", "-w", "-p", "-l", "-v"),
 	},
 }
+
+// converts "docker build --load" flag to "nerdctl build --output=type=docker".
+func handleDockerBuildLoad(_ NerdctlCommandSystemDeps, fc *config.Finch, nerdctlCmdArgs []string, index int) error {
+	if fc.DockerCompat {
+		nerdctlCmdArgs[index] = "--output=type=docker"
+	}
+
+	return nil
+}
+
+func handleBuildx(_ NerdctlCommandSystemDeps, fc *config.Finch, cmdName *string, args *[]string) error {
+	if fc == nil || !fc.DockerCompat {
+		return nil
+	}
+
+	if cmdName != nil && *cmdName == "buildx" {
+		subCmd := (*args)[0]
+		buildxSubcommands := []string{"bake", "create", "debug", "du", "imagetools", "inspect", "ls", "prune", "rm", "stop", "use", "version"}
+
+		if slices.Contains(buildxSubcommands, subCmd) {
+			return fmt.Errorf("unsupported buildx command: %s", subCmd)
+		}
+
+		logrus.Warn("buildx is not supported. using standard buildkit instead...")
+		if subCmd == "build" {
+			*args = (*args)[1:]
+		}
+		*cmdName = "build"
+	}
+	// else, continue with the original command
+	return nil
+}
+
+func handleDockerCompatInspect(_ NerdctlCommandSystemDeps, fc *config.Finch, cmdName *string, args *[]string) error {
+	if !fc.DockerCompat {
+		return nil
+	}
+
+	if *args == nil {
+		return fmt.Errorf("invalid arguments: args (null pointer)")
+	}
+
+	modeDockerCompat := `--mode=dockercompat`
+	inspectType := ""
+	sizeArg := ""
+	savedArgs := []string{}
+	skip := false
+
+	for idx, arg := range *args {
+		if skip {
+			skip = false
+			continue
+		}
+
+		if (arg == "--type") && (idx < len(*args)-1) {
+			inspectType = (*args)[idx+1]
+			skip = true
+			continue
+		}
+
+		if strings.Contains(arg, "--type") && strings.Contains(arg, "=") {
+			inspectType = strings.Split(arg, "=")[1]
+			continue
+		}
+
+		if (arg == "--size") || (arg == "-s") {
+			sizeArg = "--size"
+			continue
+		}
+
+		savedArgs = append(savedArgs, arg)
+	}
+
+	switch inspectType {
+	case "image":
+		*cmdName = "image inspect"
+		*args = append([]string{modeDockerCompat}, savedArgs...)
+	case "volume":
+		*cmdName = "volume inspect"
+		if sizeArg != "" {
+			*args = append([]string{sizeArg}, savedArgs...)
+		} else {
+			*args = append([]string{}, savedArgs...)
+		}
+	case "container":
+		*cmdName = "inspect"
+		*args = append([]string{modeDockerCompat}, savedArgs...)
+	case "":
+		*cmdName = "inspect"
+		*args = append([]string{modeDockerCompat}, savedArgs...)
+	default:
+		return fmt.Errorf("unsupported inspect type: %s", inspectType)
```

Comment:
default should not be not supported i think, it should be returned as it is and throw an error as per nerdctl

generic order that nerdctl support is

finch <type> inspect <options> identifier (this is what is supported by nerdctl)
I would say have another pr later this is fine for now for devcontainer use case